src/graph/network/msg_queue.h

```diff
@@ -55,9 +57,10 @@ struct Message {
    */
   int64_t size;
   /*!
-   * \brief aux_data pointer handler
+   * \brief To avoid NDArray been freed during the send(), we
+   * can hold a reference of current NDArray in the message.
    */
-  void* aux_handler;
+  runtime::NDArray array_handler;
```

Comment:
Shouldn't have this member (neither the `aux_handler`). Just use lambda closure to capture the NDArray.

```c++
Message msg;
msg.data = array->data;
msg.size = array->GetSIze();
msg.deallocator = [array](Message*) {};  // capture the array in the closure
```